include/picongpu/particles/atomicPhysics/kernel/UpdateElectricField.kernel

```diff
@@ -0,0 +1,126 @@
+                    DataSpace<picongpu::simDim> const cellIndex = localCellIndex + superCellCellOffset;
+
+                    // ((sim.unit.mass() * sim.unit.length())/(sim.unit.time()^2 * sim.unit.charge()))^2
+                    auto const oldEFieldVector = eFieldBox(cellIndex);
+                    auto const eFieldNormSquared = pmacc::math::l2norm2(oldEFieldVector);
+
+                    /// @note zero field also means at most zero energy use
+                    if(eFieldNormSquared <= 0)
+                        return;
+
+                    // eV * 1 = eV * sim.unit.energy()/sim.unit.energy() = (ev / sim.unit.energy()) * sim.unit.energy()
+                    // sim.unit.energy()
+                    float_X const eFieldEnergyUse
+                        = picongpu::sim.pic.get_eV() * eFieldEnergyUseCache.energyUsed(linearCellIndex);
+
+                    // sim.unit.charge()^2 * sim.unit.time()^2 / (sim.unit.mass() * sim.unit.length()^3)
+                    //  * sim.unit.length()^3
+                    // = sim.unit.charge()^2 * sim.unit.time()^2 / sim.unit.mass()
+                    constexpr float_X eps0HalfTimesCellVolume
+                        = (picongpu::sim.pic.getEps0() / 2._X) * picongpu::sim.pic.getCellSize().productOfComponents();
+
+                    /* (((sim.unit.mass() * sim.unit.length())/(sim.unit.time()^2 * sim.unit.charge()))^2
+                     *  - sim.unit.energy() / (sim.unit.charge()^2 * sim.unit.time()^2 / sim.unit.mass()))^1/2
+                     * = (((sim.unit.mass() * sim.unit.length())/(sim.unit.time()^2 * sim.unit.charge()))^2
+                     *  - (sim.unit.mass() * sim.unit.length()^2/sim.unit.time()^2
+                     *      / (sim.unit.charge()^2 * sim.unit.time()^2 / sim.unit.mass()))
+                     * )^1/2
+                     * = (((sim.unit.mass() * sim.unit.length())/(sim.unit.time()^2 * sim.unit.charge()))^2
+                     *  - (sim.unit.mass()^2 * sim.unit.length()^2/(sim.unit.time()^4 * sim.unit.charge()^2))
+                     * )^1/2
+                     * = ((sim.unit.mass()^2 * sim.unit.length()^2)/(sim.unit.time()^4 * sim.unit.charge()^2)
+                     *  - (sim.unit.mass()^2 * sim.unit.length()^2/(sim.unit.time()^4 * sim.unit.charge()^2))
+                     * )^1/2
+                     * = ((sim.unit.mass()^2 * sim.unit.length()^2)/(sim.unit.time()^4 * sim.unit.charge()^2))^1/2
+                     * = (sim.unit.mass() * sim.unit.length())/(sim.unit.time()^2 * sim.unit.charge())
+                     * = sim.unit.eField()
+                     */
+                    // sim.unit.eField()
+                    float_X const newEFieldNorm
+                        = pmacc::math::sqrt(eFieldNormSquared - eFieldEnergyUse / eps0HalfTimesCellVolume);
+
+                    eFieldBox(cellIndex) = oldEFieldVector / pmacc::math::sqrt(eFieldNormSquared) * newEFieldNorm;
```

Comment:
Are the electrons, which are created due to ionization, initialized with zero momentum?

Reply:
The are spawned co-moving with their source ion and therefore their kinetic energy comes from the ion kinetic energy, not the electric field. In theory at least, in reality the kinetic energy of the ion is not decreased since our ion masses are assumed to be constant.